biodatasets/ask_a_patient/ask_a_patient.py

```diff
@@ -0,0 +1,168 @@
+            features = schemas.kb_features
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    
+    def _split_generators(self, dl_manager):
+        dl_dir = dl_manager.download_and_extract(_URLs)
+        dataset_dir = os.path.join(dl_dir, "datasets", "AskAPatient")
+        # dataset supports k-folds 
+        splits = []
+        for split_name in [
+            datasets.Split.TRAIN,
+            datasets.Split.VALIDATION,
+            datasets.Split.TEST,
+        ]:
+            for fold_filepath in glob.glob(
+                os.path.join(dataset_dir, f"AskAPatient.fold-*.{split_name}.txt")
+            ):
+                fold_id = re.search("AskAPatient\.fold-(\d)\.", fold_filepath).group(1)
+                document_id = f"{split_name}_{fold_id}"
+                split = datasets.SplitGenerator(
+                    name=f"{split_name}_{fold_id}",
+                    gen_kwargs={
+                        "filepath": fold_filepath,
+                        "document_id": document_id
+                    },
+                )
+                splits.append(split)    
+        return splits
+
+    def _generate_examples(self, filepath, document_id):
+        with open(filepath, "r", encoding="latin-1") as f:
+            for i, line in enumerate(f):
+                id = f"{document_id}_{i}"
+                cui, medical_concept, social_media_text = line.strip().split("\t")
+                if self.config.schema == "source":
+                    yield id, {
+                        "cui": cui,
+                        "medical_concept": medical_concept,
+                        "social_media_text": social_media_text,
+                    }
+                elif self.config.schema == "bigbio_kb":
+                    text_type = "social_media_text"
+                    offset = (0, len(social_media_text))
+                    yield id, {
+                        "id": id,
+                        "document_id": document_id,
+                        "passages": [ 
+                            {
+                                "id": f"{id}_passage",
+                                "type": text_type,
+                                "text": [social_media_text],
+                                "offsets" : [offset],
+                            }
+                        ],
+                        "entities": [ 
+                            {
+                                "id": f"{id}_entity",
+                                "type": text_type,
+                                "text": [social_media_text],
+                                "offsets" : [offset],
+                                "normalized" : [
+                                    {"db_name" : "SNOMED-CT/AMT", "db_id" : cui}
```

Comment:
minor thing, but what do you think about using "|" instead of "/" here between SNOMET-CT and AMT ... I'm hoping this wont be an issue for a lot of datasets, but when we are trying to map to these databases the symbol for "or" might make a little more sense.

Reply:
use of "|" makes sense to me, have updated it